src/screens/send/components/address-book/address-book.tsx

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import { View, Image, TouchableOpacity, SafeAreaView, SectionList, Alert } from 'react-native';
+import stylesProvider from './styles';
+import { IReduxState } from '../../../../redux/state';
+import { withTheme, IThemeProps } from '../../../../core/theme/with-theme';
+import { Icon } from '../../../../components/icon';
+import { connect } from 'react-redux';
+import { smartConnect } from '../../../../core/utils/smart-connect';
+import { Text } from '../../../../library';
+import { IContactsState, IContactState } from '../../../../redux/contacts/state';
+import { selectContacts } from '../../../../redux/contacts/selectors';
+import { formatAddress } from '../../../../core/utils/format-address';
+import { translate } from '../../../../core/i18n';
+import { Blockchain } from '../../../../core/blockchain/types';
+import Swipeable from 'react-native-gesture-handler/Swipeable';
+import { deleteContact, updateContactName } from '../../../../redux/contacts/actions';
+import { confirm } from '../../../../core/utils/dialog';
+import { IAccountState } from '../../../../redux/wallets/state';
+
+export interface IReduxProps {
+    contacts: IContactsState[];
+    deleteContact: typeof deleteContact;
+    updateContactName: typeof updateContactName;
+}
+
+export interface IExternalProps {
+    blockchain: Blockchain;
+    accounts: IAccountState[];
+    onAccountSelection: (account: IAccountState) => any;
+}
+
+interface IState {
+    openedSwipeIndex: string;
+}
+
+export const mapStateToProps = (state: IReduxState, ownprops: IExternalProps) => {
+    return {
+        contacts: selectContacts(state, ownprops.blockchain)
+    };
+};
+
+const mapDispatchToProps = {
+    deleteContact,
+    updateContactName
+};
+
+export class AddressBookComponent extends React.Component<
+    IReduxProps & IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>,
+    IState
+> {
+    public walletSwipableRef: any[] = new Array();
```

Comment:
The name of the attribute is confusing. Use a more suggestive name, like `contactsSwipeableRefs`